### ISSUE TITLE
Add updateInfo method for `info` operation support

### ISSUE THREAD

**Issue by KenEucker** [kaimallea/node-imgur#106]
@kaimallea I see you're actively working on a sizeable refactor. I want to keep up with you but my need for a couple of things forces me to diverge each time you push an update to develop. Would you consider adding a method named updateInfo to support the [already existing `info` operation](https://github.com/kaimallea/node-imgur/blob/wip/src/imgur.js#L59)?

Here's the code to plug in:

```
/**
 * Update image metadata
 * @param   {string}  id - unique image id
 * @returns {promise}
 */
imgur.updateInfo = async (id, title, description) => {
  const extraFormParams = {};

  if (!id) {
      throw new Error('Invalid image ID');
  }

  if (typeof title === 'string' && title.length) {
      extraFormParams.title = title;
  }

  if (typeof description === 'string' && description.length) {
      extraFormParams.description = description;
  }

  return await imgur._imgurRequest('update', id, extraFormParams)
};
```

~I think this was code that was originally intended to be in there, as there is already supporting code for the operation in imgur.js, but there is no method that would take advantage of that operation.~

I would create a PR for this and send it to whatever branch you would like, if adding this code to the library is acceptable to you.

~REDACTED~

**Comment by KenEucker** [comment]
There are several other modifications I find myself making to support all of the functionality I am using in my application, but this is one that I felt was in line with the existing code.

If you are open to this and more, I would be happy to create a PR for adding all of it to whatever branch would be prudent.

**Comment by kaimallea** [comment]
@KenEucker hey thanks for bringing this up. Sorry for the confusion with all of my commits to various branches. 😅  Please ignore my work the in `wip` branch. It is mostly experimentation and trying to figure out a good design pattern. I don't plan on merging in any major refactoring any time soon.

Consider the `develop` branch to be the next bit of code to be merged into master (v1.0 maybe?). The changes I made in there were to make code more testable and also get rid of things like the `Q` library and take advantage of native Promises and other ES6 features. The only additional changes I plan on making to `develop` before merging it into `master` is:

1. Remove the CLI code so that this library can have a single responsibility of being a module rather than a module and CLI tool
2. Add developer tooling like commitizen and husky to enforce commits to follow conventional commit messaging, so that semantic versioning and releases can be automated via Github actions.

Basically, my hope is that making testing easier, enabling the use of ES6+ features,  and automating releases will make the repo more contributor-friendly (thank you btw 😄  )

Having said all of that, I think `updateInfo` would be a great contribution, please send a PR against the `develop` branch. 

Lastly, if you don't mind, I will add you as a contributor to the repo. I'm open to ideas/suggestions/whatever going forward.

**Comment by KenEucker** [comment]
@kaimallea this is all awesome work, thank you! I will make a PR to `develop` and I would love to keep the conversation going about more additions. Maybe I can just create individual PR's for each of the items I'm using and then we can discuss each separately?

I like the idea of removing the cli code and focusing on being a module. However, I was hoping we could retain the cli. Maybe we can create another project named `imgur-cli` and move the code to there? That way, if anyone is using the CLI right now they can continue to transition with the project. Also, if someone is motivated to invest in the CLI outside of the module I think keeping it around would benefit both projects.

I would be happy to be added as a contributor and to meet the conventions and code of conduct of this project. On that note, I don't see a conduct file or any mention of a code of conduct, but there is one thing I would like to advocate for straight away and that is to rename the default branch of the repository. I have been using the convention `production` for the last year, but I'm open to anything else that makes sense and transitions away from the use of the current term. Maybe `release` would be more appropriate?

I'll close this issue after I submit the PR. 😄

**Comment by kaimallea** [comment]
@KenEucker re: new PRs and discussions, yes, go for it!

Re: imgur-cli, sounds like the right move. you are welcome to create it under your npm username scope, no issue there.

I love the idea of the code of conduct, happy to use an existing one. I think Github has some templates somewhere.

Re: branching strategy, I had some thoughts. What do you think of "git-flow" strategy laid out in [this article](https://nvie.com/posts/a-successful-git-branching-model/)? Additionally, I found [another version](https://github.com/mobify/branching-strategy/blob/master/release-deployment.md), based on git-flow with some tweaks, already in a nice markdown format :)

**Comment by KenEucker** [comment]
> Re: imgur-cli, sounds like the right move. you are welcome to create it under your npm username scope, no issue there.

I will do this, and then plan to create a PR for replacing the CLI here with the module from there.

> I love the idea of the code of conduct, happy to use an existing one. I think Github has some templates somewhere.

I found this yesterday, and it's a good model for open collaborative works: https://www.contributor-covenant.org/

> Re: branching strategy, I had some thoughts. What do you think of "git-flow" strategy laid out in this article? 

That is the branching strategy I employ, most naturally. The markdown you linked is awesome. We can just drop that in, right?


Finally, just to be explicit, my suggestion regarding branching was to simply rename what is currently named `master` and replace it with a word that is more appropriate. Including a git-flow document for guiding contributions to the project would also be really helpful. 👍

**Comment by kaimallea** [comment]
@KenEucker re: renaming master to something more appropriate, I appreciate you recognizing and suggesting that, let's do it right now 🚀   Github has standardized on `main`, so I will do that. I think `release` can be reserved for the `release-v1.x` branch names that'll be created when we start following the git-flow model

**Comment by kaimallea** [comment]
Also, yes, let's just drop in that markdown from mobify 😄

**Comment by kaimallea** [comment]
Snippet to update name on existing local repo:

```
git branch -m master main
git fetch origin
git branch -u origin/main main
```

**Comment by KenEucker** [comment]
I started using commitizen for the work on the PR submitted, and on the "Add issue references" option I just put the ticket number, omitting the "closes" keyword. I'll get it next time!
